### PR TITLE
WIP upgrades

### PR DESCRIPTION
**Pull request by mrsimonbennett**

### CODE REVIEW

File: src/EventStore/LaravelEventStore.php [hunk 1]
```diff
@@ -95,27 +99,23 @@ public function append($id, DomainEventStream $eventStream)
...
-                             'type'        => $domainMessage->getType(),
-                         ]
-                     );
+            $this->db->table($this->eventStoreTableName)->insert($eventRow);
         } catch (\PDOException $ex) {
-            if ($ex->getCode() == 23000) {
-                throw new DuplicateAggregatePlayhead((string)$domainMessage->getId(), $domainMessage->getPlayHead());
+            if ((string) $ex->getCode() === '23000') {
+                if ($ignorePlayhead) {
+                    $eventRow['playhead'] ++;
```
Comment (mrsimonbennett) [inline src/EventStore/LaravelEventStore.php:113]: @jrdnhannah thoughts on this. Lets say you have two commands doing the same thing. they both try and save the same playhead e.g. 2 at the same time so they both go ++ (3). It happens again so they both end up at 4. One then saves at 5 and then one at 6. We now have a massive gap in playhead. Is this a race condition I need not worry about? What have you seen happen?

Comment (jrdnrc) [reply src/EventStore/LaravelEventStore.php:113]: `usleep(mt_rand(0,60))`

Comment (jrdnrc) [reply src/EventStore/LaravelEventStore.php:113]: In all seriousness though, I am not sure. Maybe try a couple of times and then exception?

Comment (mrsimonbennett) [reply src/EventStore/LaravelEventStore.php:113]: okay well I will leave for now as you don't tend to use it
and if we/others get issues using it we will work out a system thats better

**Pull request merged**